Mathlib/Topology/Instances/ENat.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2022 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.ENat.Basic
+import Mathlib.Topology.Algebra.Monoid
+import Mathlib.Topology.Instances.Discrete
+
+/-!
+# Topology on extended natural numbers
+-/
+
+open Set
+
+open scoped Topology Filter
+
+namespace ENat
+
+instance : TopologicalSpace ℕ∞ := Preorder.topology ℕ∞
+
+instance : OrderTopology ℕ∞ := ⟨rfl⟩
```

Comment:
In the `ENNReal` topology file, there is a warning not to use an emetric topology. Maybe a similar warning can be added here:

```suggestion
/-- Topology on `ℕ∞`.

Note: this is different from the `EMetricSpace` topology. The `EMetricSpace` topology has
`IsOpen {∞}`, but all neighborhoods of `∞` in `ℕ∞` contain infinite intervals. -/
instance : OrderTopology ℕ∞ := ⟨rfl⟩
```

Reply:
I would rather put that docstring on the `TopologicalSpace` instance

Reply:
Otherwise LGTM. Thanks!
bors d+